lib/activities/buildvrt.js

```diff
@@ -1,24 +1,55 @@
 "use strict";
 
-var s3Output = require("./s3_output"),
+var fs = require('fs'),
+    path = require('path');
+
+var tmp = require('tmp');
+
+var output = require("./output"),
     shell = require("./shell");
 
-module.exports = function buildVRT(files, output, callback) {
-  return s3Output(output, callback, function(err, outputFilename, done) {
+module.exports = function buildVRT(files, outputUri, callback) {
+  return output(outputUri, callback, function(err, localOutputPath, done) {
     if (err) {
       return callback(err);
     }
 
     if (files.length === 0) {
-      return done(new Error("No files provided for " + outputFilename));
+      return done(new Error("No files provided for " + localOutputPath));
     }
 
-    var args = [
-      outputFilename
-    ].concat(files);
+    if(files.length < 1000) {
+      var args = [
+        localOutputPath
+      ].concat(files);
+
+      return shell("gdalbuildvrt", args, {}, done);
+    } else {
 
-    // TODO this will fail if the list is too long
-    return shell("gdalbuildvrt", args, {}, done);
+      return tmp.tmpName({
+        postfix: ".txt"
+      }, function(err, fileList) {
+        if (err) {
+          return done(err);
+        }
+        var fd = fs.openSync(fileList, 'w');
```

Comment:
Ah yes, I think that's what I was running into. The build processed with only a partial list of the files.